engine/lib/master.go

```diff
@@ -188,6 +191,7 @@ func NewBaseMaster(
 	ctx *dcontext.Context,
 	impl MasterImpl,
 	id libModel.MasterID,
+	tp libModel.WorkerType,
```

Comment:
What about passing `WorkerType` in `dcontext.Context`

Reply:
What's the principle to pass some params in `dcontext.Context` ?

Reply:
There are no clear rules, I think passing worker type is reasonable.